### DISCUSSION TITLE
A comment on legal issues related to WCAG 2 vs APCA

### DISCUSSION THREAD

**Discussion by Myndex** [Myndex/SAPC-APCA#68] [DOCUMENTATION]
A concern that had been voiced is that "in theory" some edge cases that pass APCA may incorrectly be rejected by WCAG 2 contrast maths. The stated concern is that of a legal liability, despite the fact that APCA provides a better, actually accessible experience.

To be clear, someone attempting to bring suit against a site based soley on an automatic test, not a functional complaint, does not have a strong case—especially if they were using an alternate means that is better for actual accessibility.

It is worth noting that the existing WCAG 2 contrast specs were adopted without any form of empirical study nor peer review, and in fact did not consider the available vision and readability science available at the time. This is not a strong platform for a legal case.

I am not aware of any contrast related cases actually going to trial, and certainly none when a more accessible, and science based perceptually accurate standard was used.

We are working on an accessibility statement for the beta sites (see below), which describes the different standard being applied, and why it is functionally superior to the old standard, and we have a mass of evidence supporting this.

Unfortunately, these statements regarding "conformance concerns" have had a chilling effect in terms of the public beta test period which has slowed the testing and validation studies, and that is beyond infuriating. And the reality is that ultimate source some of these faux concerns are related to certain organizations—some that seems to have a financial stake in keeping things "the way they were" despite the clear improvements in accesibility.

APCA is stricter than WCAG 2. If the current APCA is followed correctly, the result is a more accessible and more readable experience for all. This more than satisfies the US ADA. Other nations may have different laws, but we are sourcing the majority of the beta sites in the US.

### This website is beta testing the APCA guidelines for determining text and non-text contrast. It is well known that WCAG 2 contrast maths are not relevant toward human perception, and automated testing using WCAG 2 math may incorrectly indicate errors in some color pairs. The APCA guidelines are a substantial improvement over WCAG 2 contrast criterions, and therefore compliant with the ADA and also 508 rules under the alternative methods section. Any automated testing done to determine conformance of this site needs to be conducted using a perceptually accurate method such as APCA.

Comments and questions encouraged.

**Comment by front-endian** [comment]
Thank you for writing this up! It is sad to hear that any of this has slowed the testing/development of APCA.

I agree with everything you've said, and don't have any suggested changes, but have some comments that could be worth considering,

---

My organization sells software that companies/universities run for their employees/students. For us the customers are the ones more at risk of getting sued . Convincing our internal legal/compliance people that this is an okay interpretation of the laws would be one thing, but doing so for every customer would be a challenge.

There are already many instances where different customers interpret accessibility rules in different ways. **Whether this interpretation of Section 508 is actually correct and whether it actually constitutes a liability is immaterial if a customer is dead set on WCAG 2 being their only concern.**

---

That all being said, the experiences we've had working with APCA have followed what you said: "If the current APCA is followed correctly, the result is a more accessible and more readable experience for all."

**People considering using APCA should not be concerned.** Instances where APCA passes a color that WCAG 2 would fail have been quite rare and it is easy to tweak the colors so both pass. For those who are especially paranoid, it is very easy to just measure for both both APCA and WCAG 2 (or use Bridge-PCA). We've started reworking an entire design system measuring for both and it has been easy.

**Comment by Myndex** [comment]
I just want to follow up with something that Colin said  @justcolin 

> _"... Instances where APCA passes a color that WCAG 2 would fail have been quite rare and it is easy to tweak..."_

I've touched on this in the past, but as we gather more evidence, it's important to understand that the colors that **WCAG2 contrast incorrectly _fails_**, happen to be colors that **would be very helpful** for people with **color vision deficiency**.

For instance, for the given "controversial" orange or red, WCAG2 contrast will invariably reject that color paired with white text, yet paradoxically pass that color when paired with black text. BUT because certain color vision deficient types will see that orange or red as **darker, pairing it with black text is harmful** to readability.

And as we've shown, oranges and reds paired with black are harmful for readability for all vision types, but in particular for those with certain color vision deficiencies.

Yet pairing those oranges & reds with white text **actually helps** certain color vision deficiencies, because **contrast is increased** when they see those oranges are reds as darker against the white text.

APCA passes colors against white that are helpful to CVD, while WCAG2 rejects them.

WCAG2 passes colors against black that are harmfl to CVD, while APCA rejects them.


## _Here's The Big Bombshell_

The fact that APCA passes certain oranges and reds with white text correctly, and fails those colors with black text, also means that APCA is correctly accommodating color vision deficiencies, while **WCAG2 contrast is harmful** to these color vision types. Harmful in terms of readability, and this further implies eyestrain, a lack of accommodation, and simple inaccessibility.


### Not About Design

I mentioned this because that "troll army" is out there babbling about "legal".

Well, and as per the subject of this thread:

## _ACTUAL ACCESSIBILITY IS WHAT IS LEGAL_

In reviewing court cases, WCAG2 contrast's arbitrary numbers have never won in court on _**merit**_. And arguably, they shouldn't. Most certainly, no actually accessible site has lost because it wasn't hitting WCAG two's arbitrate numbers. Keep in mind that Winn-Dixie was vacated by the 11th circuit and is **moot and not case law**.

Moreover, continuing to use something that is proven to be wrong and harmful, when a useful alternative is available, is itself something of a potential lawsuit, in that case I wouldn't expect any site that was "adhering to WCAG2" prevailing. 

**Site owners should beware:** ADA does not specify WCAG2, ADA specifies actual accessibility. **Actual accessibility is what is legal**. Even the 508 rules state very clearly that alternate methods can be used to achieve equal or better actual accessibility.

**_In other words, the individual that's been out there babbling this legal mumbo-jumbo doesn't know what the hell he's talking about. He's just a troll_.**

I've mentioned in a few articles how the web became unreadable after the passage of WCAG2 contrast specs. Before WCAG2, there was no widespread contrast problem. Sites used black text and light gray backgrounds, and were eminently more readable than sites today with their soft light gray text. You can prove this to yourself looking at the major sites on WaybackMachine.

Sites started using light gray text within a few years of WCAG to giving them the greenlight to use bad contrast for body text. **_And readability has suffered ever since._**

That, coupled with WCAG2 contrast actually **_rejecting colors that are helpful_** for color vision deficiencies while _**passing colors that are remarkably bad**_ for color vision deficiency, not to mention all vision types, any remaining trolls out there claiming this "legal" nonsense are in fact **actively obstructing actual accessibility.**


_**References:**_

- https://uxdesign.cc/better-reading-on-the-web-c943c4cfc91a
- https://tangledweb.xyz/whats-red-black-also-not-read-573b9c0a97ed
- https://tangledweb.xyz/please-stop-using-grey-text-3d3e71acfca8
- https://tangledweb.xyz/a-contrast-of-errors-373c2665d42a
- https://www.smashingmagazine.com/2022/09/realities-myths-contrast-color/

_**To the trolls:** yes, these are self-referential cites to articles I've written, because I'm the only person doing the research into why WCAG2 is so deeply flawed._

**Reply by bruce-usab** [reply]
Hello Andy, please accept my regrets that we’ve hardly corresponded since I retired in December 2024.

> APCA passes colors against white that are helpful to CVD, while WCAG2 rejects them.
> WCAG2 passes colors against black that are harmfl [sic] to CVD, while APCA rejects them.

These are two most important points in my opinion.  I’m not comfortable with “harmful” (as injurious to persons) but giving a pass to low contrast colors against black is certainly detrimental to providing and promoting accessibility.  I guess that if NNG can categorize PDFs as [Unfit for Human Consumption](https://www.nngroup.com/articles/pdf-unfit-for-human-consumption/), maybe that sort of hyperbolic language is acceptable.

My third-most concern for the WCAG 2.x algorithm is that, in my experience, people assume it’s linear.  People think 6:1 must be twice as good as 3:1 or 21:1 provides three times the contrast as 7:1.  It’s a very unintuitive scoring.  In contrast [ha], characterizing L<sup>c</sup>100 is twice as good as L<sup>c</sup>50 is … not wrong.

With regard to your [Draft Accessibility Statement for Beta Sites](#discussion-3980510):

> It is well known that WCAG 2 contrast maths are not relevant toward human perception…

The characterization of “not relevant” seems needlessly pejorative when the use of WCAG 2.x contrast maths have been a boon to accessibility.  You explain your usage in the next paragraph, but I think this is a bullet point which can stand on its own.  (Also, I don’t understand the phrase “toward human perception” – and maybe that’s on me – but it’s not plain language.)  I suggest something more objective.  Maybe:
> It is well known that WCAG 2 contrast maths are not perceptually accurate…

Or maybe:
> It is well known that WCAG 2 contrast maths are not sufficiently bias towards human perception…

I also feel compelled to comment on this bit in the accessibly statement:

> …and therefore compliant with the ADA and also 508 rules under the **alternative methods** section.

The best term to use here is [equivalent facilitation](https://www.ecfr.gov/current/title-28/section-35.203).

> Nothing in this subpart prevents the use of designs, methods, or techniques as alternatives to those prescribed, provided that the alternative designs, methods, or techniques result in substantially equivalent or greater accessibility and usability of the web content or mobile app.

I’m no lawyer, but I don’t understand why the proposed accessibility clause need be constrained to sites in beta.

**Reply by Myndex** [reply]
Hi Bruce! @bruce-usab 

Great to hear from you — no regrets needed, I hope retirement is treating you well.

These are all good catches, and I appreciate you taking the time. A few quick notes:

**On "harmful" vs "detrimental":** You're right that the language should be precise. I've been working on tightening these sorts of issues — the core point is that WCAG 2 *passes* dark-mode color pairs that fail CVD users, and that's a testable, documentable claim. I'll revise to keep it factual without the hyperbole.

**On "not relevant toward human perception":** Agreed — that phrasing was a rough draft artifact. "Not perceptually accurate" is closer to what I meant. But I like your *"It is well known that WCAG 2 contrast maths are not perceptually accurate…"* 

Related, I've actually been putting together a fairly extensive evidence review cataloging the peer-reviewed basis for that claim (Stevens, Barten, Fairchild, Legge, Peli, Buchner, etc.) — six independent lines of evidence converging on the same conclusion. So the substance is there; it's just a matter of not leading with a tone that puts people on the defensive.

**On "equivalent facilitation":** Good catch — I do use that term further down in the statement with the full §E101.2 citation, but you're right that the earlier reference should use the proper term consistently rather than the informal "alternative methods" phrasing. I'll make it consistent throughout the documentation.

**On scope beyond beta:** You're not wrong. The beta framing was originally a concession — AGWG leadership was pushing back on early adoption of APCA and wanted it throttled, so scoping the accessibility statement to beta sites was a way to not poke the bear. But it's probably time to revisit it.

I should mention — this repo is in the queue for a significant overhaul. I've been proactively reworking the documentation, evidence basis, and overall framing across the many repos and sites — letting the science lead rather than getting drawn into defensive postures (my Achilles heel). You'll see the results rolling out soon.

On the fun side — we're preparing the protan compensation extension, which further improves accommodation for red primary deficits, and a few other goodies I think you'll like.

Thanks for your continued help and support, Bruce. I genuinely wouldn't have brought this as far as it's come without your encouragement and advocacy from the very beginning.



As always, thank you for reading!


Andy
